### PR TITLE
Code refactoring and compile warning cleanup.

### PR DESCRIPTION
**Pull request by MiaYWei**
Code refactoring on human player module; 
Code modification for compile warning of computer player module; 
Added unit test files for computer player module.

### CODE REVIEW

**Comment by HongBo713** [comment]
good job 👍

**Review by HongBo713 — APPROVED** [review]

File: src/human_player_module.c [hunk 1]
```diff
@@ -83,6 +85,158 @@ bool validate_card(char *entered_value)
...
+    }
+    else
+    {
+        invalid_card_warning();
+        return RET_INVALID_CARD;
+    }
+}
+
+/**
+ * @brief human player discards a normal card
```
Comment (krithigasr) [inline src/human_player_module.c:158]: Can you explain more on the brief part here? Seems its similar to the previous function.

File: src/human_player_module.c [hunk 2]
```diff
@@ -83,6 +85,158 @@ bool validate_card(char *entered_value)
...
+    g_card_on_table.name = WILD;
+
+    //End turn
+    end_turn(HUMAN);
+    g_card_requested = false;
+    return RET_SUCCESS;
+}
+
+/**
+ * @brief Check if the player input is a action card discarding request
```
Comment (krithigasr) [inline src/human_player_module.c]: Can you change this to - "Check if the card discarded by player is an action card"?

File: src/human_player_module.c (hunk 1 above)
Comment (MiaYWei) [reply src/human_player_module.c:158]: No features change, just make the code readable by separating them into functions.

File: src/human_player_module.c (hunk 2 above)
Comment (MiaYWei) [reply src/human_player_module.c]: Done.

**Review by aditibhardwaj98 — APPROVED** [review]

**Review by krithigasr — APPROVED** [review]

**Pull request merged**